base/applications/utilman/process.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * PROJECT:         ReactOS Utility Manager (Accessibility)
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Process handling functions
+ * COPYRIGHT:       Copyright 2019 Bișoc George (fraizeraust99 at gmail dot com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "precomp.h"
+
+/* FUNCTIONS ******************************************************************/
+
+/**
+ * @IsProcessRunning
+ *
+ * Checks if a process is running.
+ *
+ * @param[in]   ProcName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the given process' name is running,
+ *     FALSE otherwise.
+ *
+ */
+BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
+{
+    BOOL bIsRunning = FALSE;
+    PROCESSENTRY32W Process = {0};
+
+    /* Create a snapshot and check whether the given process' executable name is running */
+    HANDLE hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+
+    if (hSnapshot == INVALID_HANDLE_VALUE)
+        return FALSE;
+
+    Process.dwSize = sizeof(Process);
+
+    /* Enumerate the processes */
+    if (Process32FirstW(hSnapshot, &Process))
+    {
+        do
+        {
+            if (_wcsicmp(Process.szExeFile, lpProcessName) == 0)
+            {
+                /* The process we are searching for is running */
+                bIsRunning = TRUE;
+                break;
+            }
+        }
+        while (Process32NextW(hSnapshot, &Process));
+    }
+
+    /* Free the handle and return */
+    CloseHandle(hSnapshot);
+    return bIsRunning;
+}
+
+/**
+ * @LaunchProcess
+ *
+ * Executes a process.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the process has been launched successfully,
+ *     FALSE otherwise.
+ *
+ */
+BOOL LaunchProcess(LPCWSTR lpProcessName)
+{
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+    INT_PTR Success;
+    WCHAR ExpandedCmdLine[MAX_PATH];
+
+    ExpandEnvironmentStringsW(lpProcessName, ExpandedCmdLine, ARRAYSIZE(ExpandedCmdLine));
+
+    ZeroMemory(&pi, sizeof(pi));
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    si.dwFlags = STARTF_USESHOWWINDOW;
+    si.wShowWindow = SW_SHOWNORMAL;
+
+    if (!CreateProcessW(NULL,
```

Comment:
Is there a reason for not always using `ShellExecuteW`?

Reply:
To be picky:
The OSK can be used also before logging in (in winlogon's secure desktop) and in other situations for example in Windows / NT installations where the shell (and thus, shell32) has not been installed yet.